### ISSUE TITLE
Warning: Undefined array key "parts" in Gemini provider

### ISSUE THREAD

**Issue by tg666** [inspector-apm/neuron-ai#218]
**Describe the bug**

When using the `gemini-2.5-pro` or `gemini-2.5-flash` model with a low value of the `maxOutputTokens` parameter, the `Undefined array key “parts”` warning will appear in the `src/Providers/Gemini/HandleChat.php:45` file.

Other Gemini models are passing without issue.

**To Reproduce**

```php
use NeuronAI\Providers\Gemini\Gemini;
use NeuronAI\Chat\Messages\UserMessage;

$provider = new Gemini(
     key: 'API_KEY',
     model: 'gemini-2.5-flash',
     parameters: [
         'generationConfig' => [
             'maxOutputTokens' => 5,
         ],
     ],
);

$response = $provider->chat(
    messages: [new UserMessage('ping')],
);
```

**Expected behavior**

Warning should not be produced

**Screenshots**

<img width="1159" alt="Image" src="https://github.com/user-attachments/assets/6f606c17-4879-450d-87bb-83fdd56e1622" />

**The Neuron AI version you are using:**

 - Version `1.14.28`

**Comment by ilvalerione** [comment]
An exception should be the expected behaviour anyway. No valid parts in the response can happen only if something goes wrong. It can fires a dedicated error eventually to make it more clear.

**Comment by tg666** [comment]
If it was something unexpected, I assume the API would return an error.
This, on the other hand, is the expected behavior, as the model has hit the maximum number of tokens that were defined in the request - `finishReason` is `MAX_TOKENS`.
The `parts` field is not present in the response as the model did not manage to generate anything.

I think that the code could instead behave `defensively`, as it does a little further down on line 48 - `$content['parts'][0]['text'] ?? ''`

**Comment by ilvalerione** [comment]
Return nothing if it hits finishReason = MAX_TOKENS. It seams reasonable.

**Comment by ilvalerione** [comment]
Changed to act defensively.

https://github.com/inspector-apm/neuron-ai/blob/main/src/Providers/Gemini/HandleChat.php#L45
